java-components/build-request-processor/src/main/java/com/redhat/hacbs/container/build/preprocessor/AbstractPreprocessor.java

```diff
@@ -27,36 +34,177 @@ public abstract class AbstractPreprocessor implements Runnable {
     String buildRequestProcessorImage;
 
+    @CommandLine.Option(names = "--java-version", required = true)
+    String javaVersion;
+
+    @CommandLine.Option(names = "--build-tool-version", required = true)
+    String buildToolVersion;
+
     protected enum ToolType {
         ANT,
         GRADLE,
         MAVEN,
-        SBT
+        SBT;
+
+        @Override
+        public String toString() {
+            return name().toLowerCase();
+        }
     }
 
     protected ToolType type;
 
+    /**
+     * This section creates two files within a <code>.jbs</code> subdirectory. The Containerfile is used
+     * by Konflux to initiate a build and the <code>run-build.sh</code> contains generic setup
+     * ( e.g. PATHs, directories, Maven Settings, etc) to which is appended the user build script for the main
+     * build.
+     */
     @Override
     public void run() {
         Path jbsDirectory = Path.of(buildRoot.toString(), ".jbs");
         //noinspection ResultOfMethodCallIgnored
         jbsDirectory.toFile().mkdirs();
 
+        String buildScript = System.getenv("BUILD_SCRIPT");
+        if (isEmpty(buildScript)) {
+            Log.errorf("Unable to find BUILD_SCRIPT in environment");
+            throw new RuntimeException("No BUILD_SCRIPT found");
+        }
+
+        Log.warnf("### Using tool %s with version %s and javaHome %s", type, buildToolVersion, javaVersion);
+        Log.warnf("### ENV %s", System.getenv("jvm-build-service"));
+
+        String javaHome;
+        if (javaVersion.equals("7") || javaVersion.equals("8")) {
+            javaHome = "/lib/jvm/java-1." + javaVersion + ".0";
+        } else {
+            javaHome = "/lib/jvm/java-" + javaVersion;
+        }
+
+        // TODO: Rename CACHE_URL to PROXY_URL to cover both Indy and JBS use-cases?
+        String runBuild = """
+            #!/usr/bin/env bash
+            set -o verbose
+            set -eu
+            set -o pipefail
+
+            echo "Using CACHE_URL : $CACHE_URL"
+
+            #fix this when we no longer need to run as root
+            export HOME=/root
+
+            export LANG="en_US.UTF-8"
+            export LC_ALL="en_US.UTF-8"
+            export JAVA_HOME=%s
+            export GRADLE_USER_HOME="/var/workdir/software/settings/.gradle"
+            # This might get overridden by the tool home configuration below. This is
+            # useful if Gradle/Ant also requires Maven configured.
+            export MAVEN_HOME=/opt/maven/3.8.8
+            # If we run out of memory we want the JVM to die with error code 134
+            export MAVEN_OPTS="-XX:+CrashOnOutOfMemoryError"
+            # If we run out of memory we want the JVM to die with error code 134
+            export JAVA_OPTS="-XX:+CrashOnOutOfMemoryError"
+            export %s_HOME=/opt/%s/%s
+
+            mkdir -p /var/workdir/workspace/logs /var/workdir/workspace/packages ${HOME}/.sbt/1.0 ${GRADLE_USER_HOME} ${HOME}/.m2
+            cd /var/workdir/workspace/source
```

Comment:
I wonder if I should make all of these values customisable variables so that a person could use the script locally to reproduce issues?

Reply:
Just realized, the build debug is quite tricky. In addition to getting into a container being locally or remote and to the unshare islotaed namespace. We need to provide an option to interactively connect to the container creating an image by the buildah build command, so the users can resume long running builds once thay fail in the middle eg. mvn -rf.

Reply:
A sleep can be added after the mvn command (mvn must not cause exit on failure) and `buildah run <container-id> bash` can be used to get it.

Also added a comment to the JBS-11.

Reply:
I don't think we should try and deal with debug in this iteration. We still don't know what is 'allowed' in Konflux so IMHO I don't think we should code to preempt that but establish what can be done first.

Reply:
I've changed the script generation to make it easier to override / reuse the run-build.sh.